### PR TITLE
Add golden tests

### PR DESCRIPTION
**Pull request by Acentelles**
- [X] Add Golden tests for the following phases of the compiler
  - [X] Parsing
  - [X] Typecheck (Plonk)
  - [X] Compile (Plonk)
  - [ ] Typecheck (Michelson)
  - [ ] Compile (Michelson)
  - [ ] Translate phases
- [X] Restructure `test/examples` folder
- [ ] Create "not-working" folder instead of panicking on non-working juvix files

### CODE REVIEW

**Review by cwgoes — COMMENTED** [review]
Minor comments.

File: doc/CONTRIBUTING.md [hunk 1]
```diff
@@ -113,3 +113,25 @@ there was a lot of repeat code. [See the file itself for how to mock
...
+      |- backendA
+      |- backendB
+    |- negative
+      |- backendA
+      |- backendB
+```
+
+If a change in the compiler is implemented and the output of a certain phase of the compiler is modified but correct, the following command will overwrite the existing golden output:
+```
+stack test --test-arguments "--accept"
```
[the remark this answers is not in the document]
Comment (Acentelles) [reply doc/CONTRIBUTING.md:135]: Yeah, good idea

File: doc/CONTRIBUTING.md [hunk 2]
```diff
@@ -113,3 +113,25 @@ there was a lot of repeat code. [See the file itself for how to mock
 (big comment at the top).](https://github.com/heliaxdev/juvix/blob/develop/scripts/yaml-generator.lisp)
 
 To run this after you make your changes simply type `make stack-yaml`
+
+## Golden tests
+
+Golden tests allow us to keep track of changes in all phases of the compiler: parsing, desugaring, typechecking, code generation,...
```
Comment (mariari) [inline doc/CONTRIBUTING.md:119]:
```suggestion
Golden tests allow us to keep track of changes in all phases of the compiler: parsing, desugaring, typechecking, code generation,...
```

**Review by mariari — APPROVED** [review]
Overall looks good, i appreciate the file extension names to many of the forms!

**Pull request merged**